lncfg/db.go

```diff
@@ -78,56 +79,189 @@ func (db *DB) Init(ctx context.Context, dbPath string) error {
+type boltBackendCreator func(boltCfg *kvdb.BoltConfig) (kvdb.Backend, error)
+
 // DatabaseBackends is a two-tuple that holds the set of active database
-// backends for the daemon. The two backends we expose are the local database
-// backend, and the remote backend. The LocalDB attribute will always be
-// populated. However, the remote DB will only be set if a replicated database
-// is active.
+// backends for the daemon. The two backends we expose are the graph database
+// backend, and the channel state backend.
 type DatabaseBackends struct {
-	// LocalDB points to the local non-replicated backend.
-	LocalDB kvdb.Backend
+	// GraphDB points to the database backend that contains the less
+	// critical data that is accessed often, such as the channel graph and
+	// chain height hints.
+	GraphDB kvdb.Backend
+
+	// ChanStateDB points to a possibly networked replicated backend that
+	// contains the critical channel state related data.
+	ChanStateDB kvdb.Backend
+
+	// MacaroonDB points to a database backend that stores the macaroon root
+	// keys.
+	MacaroonDB kvdb.Backend
+
+	// DecayedLogDB points to a database backend that stores the decayed log
+	// data.
+	DecayedLogDB kvdb.Backend
+
+	// TowerClientDB points to a database backend that stores the watchtower
+	// client data. This might be nil if the watchtower client is disabled.
+	TowerClientDB kvdb.Backend
+
+	// TowerServerDB points to a database backend that stores the watchtower
+	// server data. This might be nil if the watchtower server is disabled.
+	TowerServerDB kvdb.Backend
 
-	// RemoteDB points to a possibly networked replicated backend. If no
-	// replicated backend is active, then this pointer will be nil.
-	RemoteDB kvdb.Backend
+	// WalletDB is an option that instructs the wallet loader where to load
+	// the underlying wallet database from.
+	WalletDB btcwallet.LoaderOption
+
+	// Replicated indicates whether the database backends are remote, data
+	// replicated instances or local bbolt backed databases.
+	Replicated bool
 }
 
-// GetBackends returns a set of kvdb.Backends as set in the DB config.  The
-// local database will ALWAYS be non-nil, while the remote database will only
-// be populated if etcd is specified.
-func (db *DB) GetBackends(ctx context.Context, dbPath string) (
+// GetBackends returns a set of kvdb.Backends as set in the DB config.
+func (db *DB) GetBackends(ctx context.Context, chanDBPath, walletDBPath string,
+	makeMacaroonBoltDB boltBackendCreator,
+	makeDecayedLogBoltDB boltBackendCreator,
+	makeTowerClientBoltDB boltBackendCreator,
+	makeTowerServerBoltDB boltBackendCreator) (
 	*DatabaseBackends, error) {
 
-	var (
-		localDB, remoteDB kvdb.Backend
-		err               error
-	)
-
 	if db.Backend == EtcdBackend {
-		remoteDB, err = kvdb.Open(
-			kvdb.EtcdBackendName, ctx, db.Etcd,
+		etcdGraphBackend, err := kvdb.Open(
+			kvdb.EtcdBackendName, ctx,
+			db.Etcd.CloneWithSubNamespace("graphdb"),
```

Comment:
Ending all namespaces in `db`, maybe it is not necessary as everything in a database backend is data?

Reply:
This is intentional to make it more obvious this is a namespacing thing (for anyone just looking at the key names). I'd prefer to leave it that way, even if it sounds superfluous.